src/end_entity.rs

```diff
@@ -27,6 +27,9 @@ use core::convert::TryFrom;
 ///   certificate is currently valid *for use by a TLS server*.
 /// * `EndEntityCert.verify_is_valid_for_dns_name`: Verify that the server's
 ///   certificate is valid for the host that is being connected to.
+/// * `EndEntityCert.verify_is_valid_for_dns_name_or_ip`: Verify that the server's
```

Comment:
Given that (I think) rustls is the only significant user of this API, I kinda feel like it would be good enough to rename the existing method instead of adding another one. Since we rename the function and it will accept a type of a different argument, downstream users will be forced to change the code anyway, and we don't get stuck with additional API surface that will probably go unused.

What do you think?

Reply:
The main reason why I think it's important is that current users of `briansmith/webpki` won't start accepting new certificates. But I understand this does not matter much to `rustls` today. However, if a slightly different version of `webpki` merges some day and `rustls` deviated from that, it will need to adapt to the new `webpki` API.

If that is fine, I see no issue in renaming the method instead of adding a new one.

Reply:
I suppose I was assuming this would be in a semver-incompatible release, in which case it wouldn't matter, but if this is to go in a semver-compatible release this approach makes sense to me.

So you're saying this PR only makes semver-compatible changes to the public API?

Reply:
Yes, the contribution to `briansmith/webpki` was meant to be semver-compatible.